libraries/AP_Scripting/examples/ahrs-source-gps-optflow.lua

```diff
@@ -0,0 +1,184 @@
+-- this script is intended to help vehicles automatically switch between GPS and optical flow
+--
+-- setup RCx_OPTION = 90 (EKF Pos Source) to select the source (low=GPS, middle=opticalflow, high=Not Used)
+-- setup RCx_OPTION = 300 (Scripting1).  When this switch is pulled high, the source will be automatically selected
+-- setup EK3_SRCn_ parameters so that GPS is the primary source, opticalflow is secondary
+-- configure a forward or downward facing lidar with a range of more than 5m
+--
+-- When the 2nd auxiliary switch (300/Scripting1) is pulled high automatic source selection uses these thresholds:
+-- SCR_USER1 holds the threshold for rangefinder altitude:
+--     if rangefinder distance >= SCR_USER1, source1 (GPS) will be used
+--     if rangefinder distance < SCR_USER1, source2 (optical flow) will be used if innovations are below SRC_USER3 value
+-- SCR_USER2 holds the threshold for GPS speed accuracy (around 0.3 is a good choice)
+-- SCR_USER3 holds the threshold for optical flow quality (about 50 is a good choice)
+-- SCR_USER4 holds the threshold for optical flow innovations (about 0.15 is a good choice)
+
+local rangefinder_rotation = 25     -- check downward (25) facing lidar
+local source_prev = 0               -- previous source, defaults to primary source
+local sw_source_prev = -1           -- previous source switch position
+local sw_auto_pos_prev = -1         -- previous auto source switch position
+local auto_switch = false           -- true when auto switching between sources is active
+local gps_usable_accuracy = 1.0     -- GPS is usable if speed accuracy is at or below this value
+local vote_counter_max = 20         -- when a vote counter reaches this number (i.e. 2sec) source may be switched
+local gps_vs_opticalflow_vote = 0   -- vote counter for GPS vs optical (-20 = GPS, +20 = optical flow)
+
+-- play tune on buzzer to alert user to change in active source set
+function play_source_tune(source)
+  if (source) then
+    if (source == 0) then
+      notify:play_tune("L8C")       -- one long lower tone
+    elseif (source == 1) then
+      notify:play_tune("L12DD")     -- two fast medium tones
+    elseif (source == 2) then
+      notify:play_tune("L16FFF")    -- three very fast, high tones
+    end
+  end
+end
+
+-- the main update function
+function update()
+
+  -- check switches are configured
+  -- source selection from RCx_FUNCTION = 90 (EKF Source Select)
+  -- auto source from RCx_FUNCTION = 300 (Scripting1)
+  local rc_function_source = rc:find_channel_for_option(90)
+  local rc_function_auto = rc:find_channel_for_option(300)
+  if (rc_function_source == nil) or (rc_function_auto == nil) then
+    gcs:send_text(0, "ahrs-source.lua: RCx_FUNCTION=90 or 300 not set!")
+    return update, 1000
+  end
+
+  -- check rangefinder distance threshold has been set
+  local rangefinder_thresh_dist = param:get('SCR_USER1')  -- SCR_USER1 holds rangefinder threshold
```

Comment:
should use the `Parameter()` userdata for getting in the loop. You would do this in init:
```
local user1 = Parameter()
assert(user1:init('SCR_USER1'), 'get SCR_USER1 failed')
```

then call with:
`user1:get()`

Reply:
same for all the params.

Reply:
great, thanks.  changed to use this method for getting params.